docs/customize/authentication.md

```diff
@@ -89,3 +91,29 @@ Here is what the concrete configuration dictionaries generated as above look lik
 >>> helper.user_info_url
 "https://localhost:4430/auth/realms/invenio-realm/protocol/openid-connect/userinfo"
 ```
+
+*Note:* The value `remote_app["params"]["app_key"]` stores the name of the configuration
+variable holding the Keycloak client credentials.
+In case that multiple Keycloak remote apps should be registered, this variable has to be
```

Comment:
There might be an issue here. The config vars names
```
OAUTHCLIENT_KEYCLOAK_REALM_URL = helper.realm_url
OAUTHCLIENT_KEYCLOAK_USER_INFO_URL = helper.user_info_url
OAUTHCLIENT_KEYCLOAK_REMOTE_APP = helper.remote_app()
OAUTHCLIENT_KEYCLOAK_REMOTE_REST_APP = helper.remote_rest_app()
```
are hardcoded. If we have 2 keycloak providers configured, then this might not work.

WDYT?

Reply:
Each OAUTH plugging Keycloack1, Keycloack2, GITHUB ... would have a conf variable starting by the name of the module, GITHUB_APP_CREDENTIALS, KEYCLOAK_APP_CREDENTIALS. If there is another independent OAUTH provider probably those variables can be name as:

OAUTHCLIENT_KEYCLOAK1_REALM_URL = helper.realm_url
OAUTHCLIENT_KEYCLOAK1_USER_INFO_URL = helper.user_info_url
OAUTHCLIENT_KEYCLOAK1_REMOTE_APP = helper.remote_app()
OAUTHCLIENT_KEYCLOAK1_REMOTE_REST_APP = helper.remote_rest_app()

OAUTHCLIENT_KEYCLOAK2_REALM_URL = helper.realm_url
OAUTHCLIENT_KEYCLOAK2_USER_INFO_URL = helper.user_info_url
OAUTHCLIENT_KEYCLOAK2_REMOTE_APP = helper.remote_app()
OAUTHCLIENT_KEYCLOAK2_REMOTE_REST_APP = helper.remote_rest_app()

The KEYCLOAK would refer to the name of the OAUTH provided used.

Reply:
Indeed, this is what we want, but in the code it is not the case for keycloak at the moment :)

Reply:
ah yeah, true, thanks for pointing that out!
so basically we can either just alter this comment, or actually make the implementation more flexible, right?

Reply:
As discussed, we have fixed the implementation. I will update the doc in another PR.